src/index.js

```diff
@@ -1,11 +1,35 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import MapboxDraw from '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw';
+import React from "react";
+import PropTypes from "prop-types";
+import MapboxDraw from "@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw";
 
 export default class DrawControl extends React.Component {
   componentWillMount() {
-    this.draw = new MapboxDraw(this.props);
-    this.context.map.addControl(this.draw);
+    //if custom modes, add to Mapbox draw modes
+    const modes = MapboxDraw.modes;
+    if (this.props.modes) {
```

Comment:
Is this necessary? Won't it work with only `this.draw = new MapboxDraw(this.props);`, with the modes in props?

Reply:
I realized with only the custom mode, you can't be able to access the inbuilt modes that come with mapbox-gl-draw. So I thought this will be away to  use the custom mode, while still having access to the inbuilt modes

Reply:
Ah ok, got it. Maybe a simpler syntax like this:
```javascript
this.draw = new MapboxDraw({
  ...this.props,
  modes: {
    ...MapboxDraw.props,
    this.props.modes
  }
});
```

It would remove the for loop.

Reply:
doesn't seem to work..this.props.modes line shoots an error..does it expect a key value?

Reply:
Ah yeah, my bad.

```javascript
this.draw = new MapboxDraw({
  ...this.props,
  modes: {
    ...MapboxDraw.props,
    ...this.props.modes
  }
});
```